src/http.rs

```diff
@@ -86,11 +83,22 @@ impl HttpClient {
             .post(&format!("{}/entity", self.uri))
             .headers(self.headers.clone())
             .body(s)
-            .send()?
-            .text()?;
-
-        let edn_resp = Edn::from_str(&resp.replace("#inst", ""));
-        edn_resp.or(Ok(edn!({:status ":internal-server-error", :code 500})))
+            .send()?;
+
+        if resp.status().as_u16() < 300 {
+            let resp_body = resp.text()?;
+            let edn_resp = Edn::from_str(&resp_body.replace("#inst", ""));
+            edn_resp.or(Err(CruxError::BadRequestError(format!(
```

Comment:
We can use `or_else` here to avoid an unnecessary `CruxError` creation.

Reply:
This is an Error situation not an OK with bad content. Maybe in a next PR create a new error.

Reply:
This may be an alternative CruxError https://github.com/naomijub/transistor/pull/67/files

Reply:
I know, is only that using `or` creates the value eagerly and `or_else` only creates if its necessary